articles/Frontend/Architecture/coding-guide.md

```diff
@@ -102,130 +102,125 @@ return {
+- Except for base styles (reset, normalize, base and font) every rule must be scoped, by the class name of the component or a root component.
+- Try to avoid IDs in selector as much as possible and global tag selectors. Instead try to always scope the selection.
+- Selection must be done by following the semantic of the DOM ie. `.actions > button` or `.actions > [role=button]` instead of `.actions > .btn`
+- Place media queries as close to their relevant rule sets whenever possible. Don't bundle them all in a separate stylesheet or at the end of the document.
+- Don't write vendor prefixes (configure _autoprefixer_ instead).
+- Avoid unnecessary nesting and too many nesting levels.
+- Mixins and functions should be as simple as possible, serve only one purpose and be documented
+- Use variables for colors, and units that are used in multiple locations
 
 ## Best practice & Patterns
 
- JavaScript is an open language, that let's you write code in very different ways, even in different paradigms, from prototypal object oriented to functional programming. In TAO we've selected some programming paradigms and patterns over others. The goal is to bring some consistency and shared practices across the platform.
-
+JavaScript is an open language, that let's you write code in very different ways, even in different paradigms, from prototypal object oriented to functional programming. In TAO we've selected some programming paradigms and patterns over others. The goal is to bring some consistency and shared practices across the platform.
 
 ### Don't repeat Yourself
 
 > "Every piece of knowledge must have a single, unambiguous, authoritative representation within a system."
-*Andy Hunt, The Pragmatic Programmer*
+> _Andy Hunt, The Pragmatic Programmer_
 
 The simple principle will lead to code easier to maintain. So if you write the same code multiple time, think about abstractions. Abstractions doesn't need to be too high level.
 
 ### KISS
 
 > "Sometimes, the elegant implementation is just a function. Not a method. Not a class. Not a framework. Just a function."
-*John Carmack, game developer*
+> _John Carmack, game developer_
 
 If your module needs to expose a function, then your module can expose only a function, especially when there's no state, no side effect!
 
-If multiple functions serve the same purpose they can be grouped into an object serving multiple and independent *static like* methods :
+If multiple functions serve the same purpose they can be grouped into an object serving multiple and independent _static like_ methods :
 
 ```js
 //a case module util
 return {
-    capitalize : function capitalize(inputString){
-
-    },
-    camelToSnake : function camelToSnake(inputString){
-
-    }
-}
+  capitalize: function capitalize(inputString) {},
+  camelToSnake: function camelToSnake(inputString) {},
```

Comment:
**suggestion:** Use ES6+ notation in code samples
```suggestion
  capitalize(inputString) {},
  camelToSnake(inputString) {},
```

Reply:
Sure, thanks